src/haptic_buzz.c

```diff
@@ -0,0 +1,123 @@
+// Copyright 2024 Lukas Hrazky, Syler Clayton
+//
+// This file is part of the Refloat VESC package.
+//
+// Refloat VESC package is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at your
+// option) any later version.
+//
+// Refloat VESC package is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program. If not, see <http://www.gnu.org/licenses/>.
+
+#include "haptic_buzz.h"
+
+void haptic_buzz_configure(HapticBuzz *haptic_buzz, RefloatConfig *float_conf) {
+    haptic_buzz->haptic_buzz_intensity = float_conf->haptic_buzz_intensity;
+    haptic_buzz->haptic_buzz_min = float_conf->haptic_buzz_min;
+    haptic_buzz->haptic_buzz_duty = float_conf->haptic_buzz_duty;
+    haptic_buzz->haptic_buzz_hv = float_conf->haptic_buzz_hv;
+    haptic_buzz->haptic_buzz_lv = float_conf->haptic_buzz_lv;
+    haptic_buzz->haptic_buzz_temp = float_conf->haptic_buzz_temp;
+}
+
+void haptic_buzz_reset(HapticBuzz *haptic_buzz, float current_time) {
+    haptic_buzz->haptic_tone_in_progress = false;
+    haptic_buzz->haptic_timer = current_time;
+    haptic_buzz->applied_haptic_current = 0;
+}
```

Comment:
The function doesn't completely initialize the structure. The remaining variables are initialized on lines 117-120, relying on no buzz being played on the first call. Please make the initialization here complete for clarity and simplicity.

Reply:
Done